fortitude/src/check.rs

```diff
@@ -1048,14 +1052,23 @@ pub fn check(args: CheckArgs, global_options: &GlobalConfigArgs) -> Result<ExitC
             }
         });
 
-    let mut all_diagnostics = diagnostics_per_file
-        .fold(Diagnostics::default, |all_diagnostics, file_diagnostics| {
-            all_diagnostics + file_diagnostics
-        })
-        .reduce(Diagnostics::default, |a, b| a + b);
+    let (mut all_diagnostics, checked_files) = diagnostics_per_file
```

Comment:
Am I right in thinking that `checked_files` includes only files that were successfully scanned? If so, could we propagate this to `write_summary_text` so we can get the number of checked files right when `IoError` is being ignored? It would also be nice if we could propagate the duration so the message could say "checked x files in y seconds", but that could wait for a later issue/PR.

Reply:
No, this is essentially a duplication of the same calculation in `write_summary_text`, it's just the same as what ruff reports at this stage (that is, the file was scanned, and the error was "couldn't be read")